packages/provider/src/json-rpc-provider.ts

```diff
@@ -93,14 +93,25 @@ export class JsonRpcProvider implements AbstractProvider {
         ]
       : this.rpcUrl
 
-    const routedRpcUrl = `${finalRpcUrl}${route}`
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    const url = new URL(route, finalRpcUrl);
+    if (url.username || url.password) {
+      const credential =
+        Buffer.from(`${url.username}:${url.password}`).toString('base64');
```

Comment:
I'm not positive how this package is handling polyfills, so depending on that `Buffer` could potential cause problems.

Reply:
Ok, I imported the "buffer" module as another module (e.g. [signer](https://github.com/pokt-network/pocket-js/blob/main/packages/signer/src/key-manager.ts#L3)) in this repo does.